backend/modules/simulation/agents/grading_agent.py

```
@@ -151,6 +153,77 @@ def _format_overall_feedback(self, result: OverallGradingResult) -> str:
+            for col in expected_columns:
+                if col.lower() in output_text.lower():
+                    results["columns_found"].append(col)
+                else:
+                    results["missing_columns"].append(col)
+
+        # Check minimum row count
+        expected_rows_min = automated_checks.get("expected_rows_min")
+        if expected_rows_min is not None and output_text:
+            # Count non-empty lines in output as a rough row proxy
+            data_lines = [
+                l for l in output_text.split("\n")
```

Ruff (0.15.4)

[error] Ambiguous variable name: `l`

(E741)